core/src/main/scala/org/broadinstitute/dsde/rawls/submissions/SubmissionsService.scala

```diff
@@ -612,15 +613,36 @@ class SubmissionsService(
       header = SubmissionValidationHeader(methodConfig.rootEntityType, methodConfigInputs, entityProvider.entityStoreId)
 
       workspaceExpressionResults <- evaluateWorkspaceExpressions(workspaceContext, gatherInputsResult)
-      submissionParameters <- entityProvider.evaluateExpressions(
-        ExpressionEvaluationContext(submissionRequest.entityType,
-                                    submissionRequest.entityName,
-                                    submissionRequest.expression,
-                                    methodConfig.rootEntityType
-        ),
-        gatherInputsResult,
-        workspaceExpressionResults
-      )
+      submissionParameters <-
+        if (submissionRequest.entityNames.isEmpty) {
+          entityProvider.evaluateExpressions(
+            ExpressionEvaluationContext(submissionRequest.entityType,
+                                        submissionRequest.entityName,
+                                        submissionRequest.expression,
+                                        methodConfig.rootEntityType
+            ),
+            gatherInputsResult,
+            workspaceExpressionResults
+          )
+        } else {
+          submissionRequest.entityNames match {
+            case Some(entityNames) =>
+              Future
+                .sequence(entityNames.map { entityName =>
+                  entityProvider.evaluateExpressions(
+                    ExpressionEvaluationContext(submissionRequest.entityType,
+                                                Some(entityName),
+                                                submissionRequest.expression,
+                                                methodConfig.rootEntityType
+                    ),
+                    gatherInputsResult,
+                    workspaceExpressionResults
+                  )
+                })
+                .map(_.flatten.to(LazyList))
```

Comment:
would it be helpful to have a new method in `EntityProvider` to evaluate expressions across multiple entities? Else, this is going to execute queries for each entity in the loop.